crates/router/src/core/payments/operations/payment_update.rs

```diff
@@ -193,6 +198,21 @@ impl<F: Send + Clone, Ctx: PaymentMethodRetrieve>
         )
         .await?;
 
+        let payment_method_billing = helpers::create_or_update_address_for_payment_by_request(
+            db,
+            request.billing.as_ref(),
```

Comment:
Shouldn't we take this from payment_method_data.billing?

Reply:
Good catch, this must be taken from request.payment_method_data. will change it. Will also add test cases covering this